src/protocols/secure_channel/PASESession.cpp

```diff
@@ -470,6 +471,7 @@ CHIP_ERROR PASESession::HandlePBKDFParamResponse(System::PacketBufferHandle && m
         if (tlvReader.Next() != CHIP_END_OF_TLV)
         {
             SuccessOrExit(err = DecodeMRPParametersIfPresent(TLV::ContextTag(5), tlvReader));
+            mExchangeCtxt->GetSessionHandle()->AsUnauthenticatedSession()->SetRemoteMRPConfig(mRemoteMRPConfig);
```

Comment:
Why do we have this extra `if (mHavePBKDFParameters)` case?
Isn't `mHavePBKDFParameters = false` always, so this seems to be a dead code path awaiting some unknown future support.  I'd suggest we remove `mHavePBKDFParameters` and simplify the TLV parsing to be a single linear logic block.

Reply:
will do it in the following pr